### PR TITLE
Add batch assignments to editors

### LINKED ISSUE #4527: Improve workflow for assigning many articles to a single editor
See https://phabricator.wikimedia.org/T284760

### PR DESCRIPTION
**Pull request by TheTrio**
Fixes #4527 

https://user-images.githubusercontent.com/10794178/224190227-17ee9cc6-fe97-49e8-ab68-84821fc89c7d.mp4

### CODE REVIEW

File: app/assets/javascripts/components/common/AssignCell/AssignButton.jsx [hunk 1]
```diff
@@ -243,14 +243,24 @@ export class AssignButton extends React.Component {
...
     const title = e.target.value;
-    const assignment = CourseUtils.articleFromTitleInput(title);
-    const language = assignment.language || this.state.language;
-    const project = assignment.project || this.state.project;
+    const articlesTitles = [];
+    let language;
+    let project;
+    title.split('\n').forEach((articleTitle) => {
+      if (!articleTitle) {
+        articlesTitles.push('');
```
Comment (ragesoss) [inline app/assets/javascripts/components/common/AssignCell/AssignButton.jsx:251]: What's this for?

File: spec/features/multiwiki_assignment_spec.rb [hunk 2]
```diff
@@ -48,6 +47,35 @@
...
+      visit "/courses/#{course.slug}/students/articles"
+      first('.student-selection .student').click
+
+      button = first('.assign-button')
+      expect(button).to have_content 'Assign/remove an article'
+      button.click
+
+      within('#users') do
+        first('textarea').set(
+          "Terre\nhttps://fr.wikipedia.org/wiki/Anglais"
```
Comment (ragesoss) [inline spec/features/multiwiki_assignment_spec.rb:61]: Nice.

File: app/assets/javascripts/components/common/AssignCell/AssignButton.jsx (hunk 1 above)
Comment (TheTrio) [reply app/assets/javascripts/components/common/AssignCell/AssignButton.jsx:251]: the pushing of the empty string? That's to make sure that user entered new lines are preserved - without it, pressing enter would do essentially nothing.

Comment (ragesoss) [reply app/assets/javascripts/components/common/AssignCell/AssignButton.jsx:251]: Maybe add a comment to that effect.

**Pull request merged**